### PR TITLE
chore(docs): add APIError to debugging docs

### PR DESCRIPTION
**Pull request by noahdietz**
Expand the existing Go debugging docs to include the "new" `apierror.APIError` and the potential for `googleapi.Error` REST transport errors. 

cc: @summer-ji-eng

### CODE REVIEW

**Comment by codyoss** [comment]
Do you think we still need the old gRPC docs? I wonder if we try to start guiding users to just the new way. Thoughts?

**Comment by noahdietz** [comment]
> Do you think we still need the old gRPC docs? I wonder if we try to start guiding users to just the new way. Thoughts?

I think we want to keep it for now. If they want to inspect the gRPC status Code, they will want to unwrap it, same with `googleapi.Error` and the HTTP status code. I'm hesitant to add a `Code` API to the `APIError` because the numbers are different between transports.

File: doc.go [hunk 1]
```diff
@@ -165,9 +165,35 @@ For HTTP logging, set the GODEBUG environment variable to "http2debug=1" or "htt
...
-`grpc.Status`. Converting your errors to this type can be a useful to get
-more information about what went wrong while debugging.
+Most of the errors returned by the generated clients are wrapped in an
+`apierror.APIError` (https://pkg.go.dev/github.com/googleapis/gax-go/v2/apierror)
+and can be further unwrapped into a `grpc.Status` or `googleapi.Error` depending
+on the transport used to make the call (gRPC or REST). Converting your errors to
+these types can be a useful way to get more information about what went wrong
+while debugging.
+
+Converting to an `apierror.APIError` gives access to specific details in the
```
Comment (quartzmo) [inline doc.go:175]: "Converting" in this context sounds to me like an action that can taken by the reader.

File: doc.go [hunk 2]
```diff
@@ -165,9 +165,35 @@ For HTTP logging, set the GODEBUG environment variable to "http2debug=1" or "htt
...
+error. The transport-specific errors can still be unwrapped using the
+`apierror.APIError`.
+ if err != {
```
[the remark this answers is not in the document]
Comment (noahdietz) [reply doc.go:178]: Yep! Copy pasta issue - it was already like this! https://pkg.go.dev/cloud.google.com/go#hdr-Inspecting_errors

File: doc.go (hunk 1 above)
Comment (noahdietz) [reply doc.go:175]: I'm now sure what you mean - it is something that the user would do?

Comment (quartzmo) [reply doc.go:175]: Right, to me this sounds like something the user would do. So maybe just drop "Converting to an ":

```
`apierror.APIError` gives access to specific details in the
```

Comment (noahdietz) [reply doc.go]: Oh oh gotchya. Done!

**Review by codyoss — APPROVED** [review]

**Pull request merged**